gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/ContextAwareMetricFactory.java

```diff
@@ -51,10 +54,28 @@
    */
   public T newMetric(MetricContext context, String name);
 
-  default public T newMetric(MetricContext context, String name, Object... args) {
+  default public T newMetric(ContextAwareMetricFactoryArgs args) {
     return null;
   }
 
+  @Data
+  @AllArgsConstructor
+  class ContextAwareMetricFactoryArgs {
```

Comment:
Let's move this class to a new file. We can declare its subclasse as inner class